### PR TITLE
doc: add hypervisor kconfig option reference

### PR DESCRIPTION
**Pull request by dbkinder**
Modify the Zephyr project script for creating documentation from Kconfig
files to generate the hypervisor option configuration docs.

Signed-off-by: David B. Kinder <david.b.kinder@intel.com>

### CODE REVIEW

**Comment by NanlinXie** [comment]
Can one of the admins verify this patch?

**Comment by dbkinder** [comment]
@junjiemao1 This should get us the hypervisor config option documentation (mention in #411)

**Comment by dbkinder** [comment]
It's also worth noting the Zephyr project uses its own version of kconfiglib and the config documentation generation script relies on this version so it included here too.

**Comment by junjiemao1** [comment]
@ulfalizer welcome :)

@dbkinder I've tried with the vanilla Kconfiglib from Pypi. It also works and there is no differences among the generated files.

File: doc/Makefile [hunk 1]
```diff
@@ -14,9 +14,10 @@ SPHINXPROJ    = "Project ACRN"
 SOURCEDIR     = .
 BUILDDIR      = _build
 
+ACRN_BASE     = "`pwd`/../.."
```
Comment (gvancuts) [inline doc/Makefile]: Could we use `ACRN_BASE = $(CURDIR)/../..` here instead? (or `$(shell pwd)/../..`)

**Comment by dbkinder** [comment]
I've incorporated your comments @ulfalizer @gvancuts @junjiemao1 Thanks!

**Comment by dbkinder** [comment]
@ulfalizer looks like genrest.py (when run in the CI environment) doesn't like non-ascii characters:

```
make[1]: Entering directory '/workspace/acrn-hypervisor-premerge-build-nuc/acrn-hypervisor/doc'
Traceback (most recent call last):
  File "scripts/genrest.py", line 377, in <module>
    write_kconfig_rst()
  File "scripts/genrest.py", line 97, in write_kconfig_rst
    write_sym_rst(sym, out_dir)
  File "scripts/genrest.py", line 123, in write_sym_rst
    kconfig_definition_rst(sym))
  File "scripts/genrest.py", line 373, in write_if_updated
    f.write(s)
UnicodeEncodeError: 'ascii' codec can't encode character '\u2192' in position 335: ordinal not in range(128)
make[1]: Leaving directory '/workspace/acrn-hypervisor-premerge-build-nuc/acrn-hypervisor/doc'
make[1]: *** [Makefile:48: kconfig] Error 1
make: *** [Makefile:41: doc] Error 2
```

**Comment by ulfalizer** [comment]
@dbkinder 
Only meant safe in the font sense there. ;)

Yeah, should probably just force UTF-8 in `genrest.py`. Kconfiglib already defaults to UTF-8.

Might be a good idea to switch to some UTF-8 locale in the CI environment too (`C.UTF-8` is usually available). The root cause of the problem is that that the environment has `LC_CTYPE` set to `C`, which tells Python to use ASCII as the encoding. Other tools besides Python will check that setting as well.

Related Python PEP (it's been enough of a pain that they decided to add a workaround for Python 3.7): https://www.python.org/dev/peps/pep-0538/

**Comment by dbkinder** [comment]
@ulfalizer thanks for the info, should we tweak genrest.py in Zephyr as well, just in case?

**Comment by ulfalizer** [comment]
@dbkinder 
Can't hurt. Want to make a PR?

**Pull request merged**